### PR TITLE
[JUJU-1453] Test charm cleanup

### PR DESCRIPTION
**Pull request by tlm**
Continued work that is stable for cleaning up charms in our testing infra.

- Upgrading charms for charmcraft/charmhub support.
- Adding LTS ubuntu releases including jammy support.
- Adding support for other arch's besides amd64
- Moving tests to use cloud charms from Charmhub and not performing local deploys unless that is the test's purpose.

## Checklist

 - [x] Requires a [pylibjuju](https://github.com/juju/python-libjuju) change
 - [x] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR
 - [x] Added or updated [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) related to packages changed
 - [x] Comments answer the question of why design decisions were made

## QA steps

- Verify uploaded charms in Charmhub
- Run affected tests suits locally for successful test run.

## Documentation changes

N/A

## Bug reference

N/A

### CODE REVIEW

File: acceptancetests/repository/bundles-lxd-profile-upgrade.yaml [hunk 1]
```diff
@@ -1,15 +1,15 @@
 series: bionic
 applications:
   lxd-profile:
-    charm: cs:~juju-qa/bionic/lxd-profile-without-devices-2
+    charm: juju-qa-lxd-profile-without-devices
```
Comment (wallyworld) [inline acceptancetests/repository/bundles-lxd-profile-upgrade.yaml:4]: we need to ensure we retain in a test bundle somewhere (if not already thought about):
- revision pinning
- deploy from cs

File: api/agent/uniter/application.go [hunk 2]
```diff
@@ -116,6 +117,11 @@ func (s *Application) CharmURL() (string, bool, error) {
 // SetStatus sets the status of the application if the passed unitName,
 // corresponding to the calling unit, is of the leader.
 func (s *Application) SetStatus(unitName string, appStatus status.Status, info string, data map[string]interface{}) error {
+	fmt.Println("we are here")
+	fmt.Println("we are here")
+	fmt.Println("we are here")
+	fmt.Println("we are here")
```
Comment (wallyworld) [inline api/agent/uniter/application.go:123]: but i am there

File: api/agent/uniter/application.go [hunk 3]
```diff
@@ -116,6 +117,11 @@ func (s *Application) CharmURL() (string, bool, error) {
...
+	spew.Dump(appStatus)
```
Comment (wallyworld) [inline api/agent/uniter/application.go:124]: yikes

File: testcharms/charms/README.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Charms Directory
+
+## Introduction
+This directory is responsible for holding all of the source code for the various
+charms that Juju uses during acceptance/bash tests. Charms in this directory are
```
Comment (wallyworld) [inline testcharms/charms/README.md:5]: just "acceptance tests"

**Review by wallyworld — APPROVED** [review]
A couple of cleanups

File: acceptancetests/repository/bundles-lxd-profile-upgrade.yaml (hunk 1 above)
Comment (tlm) [reply acceptancetests/repository/bundles-lxd-profile-upgrade.yaml:4]: i'll add it into the todo. I'll construct a test specifically for this purpose.

**Pull request merged**